src/cargo-kani/src/main.rs

```diff
@@ -83,17 +99,37 @@ fn standalone_main() -> Result<()> {
     if let Some(restriction) = outputs.restrictions {
         ctx.apply_vtable_restrictions(&linked_obj, &restriction)?;
     }
-    ctx.run_goto_instrument(&linked_obj, &linked_obj, &ctx.args.function)?;
-
-    if ctx.args.visualize {
-        ctx.run_visualize(&linked_obj, "report")?;
-    } else {
-        let result = ctx.run_cbmc(&linked_obj)?;
-        if result == VerificationStatus::Failure {
-            // Failure exit code without additional error message
-            drop(ctx);
-            std::process::exit(1);
+
+    let metadata = ctx.collect_kani_metadata(&[outputs.metadata])?;
```

Comment:
Can we move the duplicated code to a separate function?

Reply:
I've tried to avoid having any functions that are specific to `kani` or `cargo-kani` (except these two of course) because I think these are generally design errors. You'll notice that there are several small variations between these two functions, and trying to abstract those differences out will generally make the code worse, not better.

A small amount of not-really-duplication is better than trying to create a function with a half dozen parameters that only gets called 2 ways. The old python was full of that, and it's just code that only gets worse as people add to it.

That said, I think there IS a little bit I can factor out here, I just need to fix how we do report generation first since I actually broke that...

Reply:
That makes sense. Thanks